tests/ui/feature-gates/feature-gate-never_patterns.rs

```diff
@@ -0,0 +1,18 @@
+// Check that never patterns require the feature gate.
+use std::ptr::NonNull;
+
+enum Void {}
+
+fn main() {
+    let res: Result<u32, Void> = Ok(0);
+    let (Ok(_x) | Err(&!)) = res.as_ref();
+    //~^ ERROR `!` patterns are experimental
+    //~| ERROR: is not bound in all patterns
+
+    unsafe {
+        let ptr: *const Void = NonNull::dangling().as_ptr();
+        match *ptr {
+            ! => {} //~ ERROR `!` patterns are experimental
```

Comment:
Can you add a test like:

```
#[cfg(FALSE)]
fn test() {
  match () {
    ! => {}
  }
}
```

That checks that we're properly feature-gating the pattern even in a cfg'd out context?